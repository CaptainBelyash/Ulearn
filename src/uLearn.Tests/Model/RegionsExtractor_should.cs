using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using uLearn.Model.Blocks;

namespace uLearn.Model
{
	[TestFixture]
	public class RegionsExtractor_should
	{
		private static readonly DirectoryInfo dir = new DirectoryInfo("tests");

		private static RegionsExtractor GetExtractor(string file = "OverloadedMethods.cs", string langId = "cs")
		{
			var code = dir.GetFile(file).ContentAsUtf8();
			return new RegionsExtractor(code, langId);
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void ExtractRegion()
		{
			var extractor = GetExtractor();
			Approvals.Verify(extractor.GetRegion(new Label { Name = "methods_1_and_2" }));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void ExtractMethod()
		{
			var extractor = GetExtractor();
			Approvals.Verify(extractor.GetRegion(new Label { Name = "Method" }));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void ExtractMany()
		{
			var extractor = GetExtractor();
			Approvals.VerifyAll(extractor.GetRegions(new[]
			{
				new Label { Name = "Method" },
				new Label { Name = "Main", OnlyBody = true },
				new Label { Name = "methods_1_and_2" },
				new Label { Name = "methods_2_and_3" }
			}), "regions");
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void ExtractPriority()
		{
			var extractor = GetExtractor();
			Approvals.Verify(extractor.GetRegion(new Label { Name = "Region" }));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void ExtractManyNoCs()
		{
			var extractor = GetExtractor(langId: "");
			Approvals.VerifyAll(extractor.GetRegions(new[]
			{
				new Label { Name = "Method" },
				new Label { Name = "Main", OnlyBody = true },
				new Label { Name = "methods_1_and_2" },
				new Label { Name = "methods_2_and_3" }
			}), "regions");
		}
	}
}