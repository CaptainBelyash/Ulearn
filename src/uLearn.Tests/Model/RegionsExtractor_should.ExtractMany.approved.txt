regions[0] = public string Method()
{
	return "a";
}

public string Method(int a)
{
	return a.ToString("0");
}

private string Method(string a)
{
	return a;
}
regions[1] = return;
regions[2] = public string Method()
{
	return "a";
}

//region methods_2_and_3

public string Method(int a)
{
	return a.ToString("0");
}
regions[3] = public string Method(int a)
{
	return a.ToString("0");
}

//endregion methods_1_and_2

private string Method(string a)
{
	return a;
}
